### PR TITLE
BIG update

### PR DESCRIPTION
**Pull request by miguelcobain**
* update to ember-cli 2.7.0
* fixes bugs in some browsers
  * perhaps the most important change is that this no longer uses the `transitionend` event. tl;dr it is useless and buggy. We now use timeouts according to the transition times. and more interesting info: https://github.com/facebook/react/pull/4561
* adds some tests
* **A LOT** of clean up:
  * Should have perf improvements, no more queues, etc
  * codebase much easier to follow
  * removed unnecessary service
  * better api: more aligned with react/angular namings. Less confusing.
  * the process was reviewed step by step
* adds `ember-suave` for stylish code

### CODE REVIEW

**Comment by miguelcobain** [comment]
@snewcomer @jordpo It looked like you were using this addon in your projects.

Could you please test this PR?
It seems to be working great for us at ember-paper.

**Comment by miguelcobain** [comment]
also, @peec to eventually release this I will need rights to npm on this project.
Let's not publish it right away, though.

**Comment by miguelcobain** [comment]
I'm loving this update so far and I think we should release it.
@peec I can do that for you, but I will need to be a collaborator at npm on this project.

Let me know if I can help any further.

btw, tests are passing in ember-canary which has glimmer 2. :smile:

**Comment by peec** [comment]
@miguelcobain Thanks very much, currently my time is limited with other projects. I added you as owner also to npm!

File: addon/mixins/transition-mixin.js [hunk 1]
```diff
@@ -26,203 +75,164 @@ export default Mixin.create({
     return this.get('transitionClasses').join(' ');
   }),
 
-  addClass(className, $element) {
+  addClass(className, element) {
     if (!this.get('isDestroying')) {
-      run(() => {
-        this.get('transitionClasses').addObject(className);
-      });
+      this.get('transitionClasses').addObject(className);
```
Comment (snewcomer) [inline addon/mixins/transition-mixin.js]: @miguelcobain I think this `run` is required?? I was getting errors related to the run loop in testing.  Thoughts?

Comment (snewcomer) [inline addon/mixins/transition-mixin.js]: https://github.com/peec/ember-css-transitions/pull/14

Other ppl were having the same problem.

Comment (snewcomer) [inline addon/mixins/transition-mixin.js]: And if so, this line would need it too.

https://github.com/peec/ember-css-transitions/blob/278c6699f088b2289c4cf781328cf54d69575a63/addon/mixins/transition-mixin.js#L232

Comment (miguelcobain) [inline addon/mixins/transition-mixin.js]: @snewcomer thanks for testing. What error came up exactly?

Comment (miguelcobain) [inline addon/mixins/transition-mixin.js]: I didn't find any problems with ember-paper's tests.

Comment (miguelcobain) [inline addon/mixins/transition-mixin.js]: @snewcomer will release this. let me know if you find any problems.

**Pull request merged**